src/pricecypher/datasets.py

```diff
@@ -176,6 +179,26 @@ def get_transactions(
         ]
         # Find all scope value filters to be sent to the dataset service.
         filters = self._find_scope_value_filters(columns_with_values)
+
+        # If additional filters for scopes that may not be added to the transactions
+        # are sepcified, get the scope_value_id for the scope_id and scope_values
+        # and extend the filters list
+        scope_ids_to_be_filtered = [scope['scope_id'] for scope in filter_scopes]
+        scope_values_to_be_filtered = [scope['values'] for scope in filter_scopes]
+
+        filters_additional = []
+
+        for id, values in zip(scope_ids_to_be_filtered, scope_values_to_be_filtered):
+            filters_additional.extend(self.get_scope_values(dataset_id, id).where_in(values).pluck('id'))
```

Comment:
Can you put this in a separate function please?

Reply:
done!